src/helpers/canonical-helper.php

```diff
@@ -0,0 +1,49 @@
+<?php
+/**
+ * A helper object for the canonical meta tag.
+ *
+ * @package Yoast\WP\Free\Helpers
+ */
+
+namespace Yoast\WP\Free\Helpers;
+
+/**
+ * Class Canonical_Helper
+ */
+class Canonical_Helper {
+
+	/**
+	 * @var Url_Helper
+	 */
+	protected $url;
+
+	/**
+	 * Sets the required helpers.
+	 *
+	 * @required
+	 * @codeCoverageIgnore
+	 *
+	 * @param Url_Helper $url The url helper.
+	 */
+	public function set_helpers( Url_Helper $url ) {
+		$this->url = $url;
+	}
+
+	/**
+	 * Applies global settings and does clean-up for the robots settings for all types of pages.
+	 *
+	 * @param string $canonical The robots settings.
+	 *
+	 * @return string The page's canonical.
+	 */
+	public function after_generate( $canonical ) {
```

Comment:
`after_generate` is a not so helpful name? What does it exactly?  I would name it `make_absolute` or  something like that. WordPress devs might consider `ensure_absolute_url`

Seeing this code and with my knowledge about the canonical that always should be a url, I would move this to the url helper. This makes it reusable for the images value objects that contains the exact two lines on 42 and 43